src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -517,6 +517,9 @@ public static bool IsSimpleIdentCh(char ch)
         // Returns true if the specified character is a new line character.
         public static bool IsNewLineCharacter(char ch) => ch == '\n';
 
+        // Returns true if the specified character is a semicolon character.
+        public static bool IsSemicolonCharacter(char ch) => ch == ';';
```

Comment:
Don't compare to semicolon, generally. It's `;;` in some locales as well. 

 TexlLexer.TryGetPunctuator(str, out tid) && tid = TokKind.Semicolon